main.tf

```diff
@@ -1,15 +1,34 @@
+locals {
+  values = yamlencode({
+    "awsRegion" : data.aws_region.current.name,
+    "autoDiscovery" : {
+      "clusterName" : var.cluster_name
+    },
+  })
+}
+
 data "aws_region" "current" {}
 
-resource "helm_release" "self" {
+data "utils_deep_merge_yaml" "values" {
   count = var.enabled ? 1 : 0
+  input = compact([
+    local.values,
+    var.values
+  ])
+}
```

Comment:
We don't need deep merge data source because overriding default chart values aren't necessary (right now there is no use case)